app/code/Magento/AdminNotification/Block/Grid/Renderer/Actions.php

```diff
@@ -40,16 +45,16 @@ public function __construct(
      * @param   \Magento\Framework\DataObject $row
      * @return  string
      */
-    public function render(\Magento\Framework\DataObject $row)
+    public function render(DataObject $row) : string
```

Comment:
Unfortunately, we are not able to change public method signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
The [Technical Guidelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html?itm_source=devdocs&itm_medium=search_page&itm_campaign=federated_search&itm_term=return%20type#1-basic-programming-principles) do describe that:

```
...
1.2. Explicit return types MUST BE declared on functions.

1.3. Type hints for scalar arguments SHOULD be used.

...
```

For me then, it seems to be fine to have a strict type?

Reply:
@DavidLambauer , yes, it's true for the new code, but for existing code, it's backward incompatible change, because every developer who extended this class and has own render method will get an error because method signature is changed.